### ISSUE TITLE
crash in AttributionFetcher, ROT13 - affecting aobut 5% users in the field.

### ISSUE THREAD

**Issue by quantamrhino** [RevenueCat/purchases-ios#1685]
- [ yes] I have updated Purchases SDK to the latest version
- [yes ] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-ios/blob/main/Development/CONTRIBUTING.md)
- [ yes] I have searched the [Community](https://community.revenuecat.com)
- [ yes] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [yes ] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-ios/issues)

**Describe the bug**
Just a crash happening in the field, on ios 15. It's hitting about 5% users.

1. Environment
   1. Platform: iOS  
   2. SDK version: 4.5.2
   3. StoreKit 2 enabled (Y/N): not sure
   4. OS version: 15.5
   5. Xcode version: 13
   6. How widespread is the issue. Percentage of devices affected. 5-7%
   
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue - None available
3. Steps to reproduce, with a description of expected vs. actual behavior - not sure. just happens at random points in the app - not in the purchase flow.
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)5. 
```
Crashed: com.apple.root.user-initiated-qos.cooperative
0  MyApp                    0x326f08 specialized Collection.map<A>(_:) + 4341051144 (<compiler-generated>:4341051144)
1  MyApp                    0x327324 specialized static ROT13.string(_:) + 93 (String+Extensions.swift:93)
2  MyApp                    0x30fa24 specialized AttributionFetcher.trackingProxy.getter + 4340955684 (<compiler-generated>:4340955684)
3  MyApp                    0x30fdec specialized AttributionFetcher.fetchAuthorizationStatus.getter + 141 (AttributionFetcher.swift:141)
4  MyApp                    0x395bd4 SubscribersAPI.post(receiptData:appUserID:isRestore:productData:presentedOfferingIdentifier:observerMode:subscriberAttributes:completion:) + 4341504980 (<compiler-generated>:4341504980)
5  MyApp                    0x3c0124 closure #2 in closure #1 in PurchasesOrchestrator.syncPurchases(receiptRefreshPolicy:isRestore:completion:) + 98 (Backend.swift:98)
6  MyApp                    0x3c4594 partial apply for closure #2 in closure #1 in PurchasesOrchestrator.syncPurchases(receiptRefreshPolicy:isRestore:completion:) + 4341695892 (<compiler-generated>:4341695892)
7  MyApp                    0x3e5f44 closure #1 in TransactionsManager.customerHasTransactions(receiptData:completion:) + 32 (TransactionsManager.swift:32)
8  libswift_Concurrency.dylib     0x3b7cc swift::runJobInEstablishedExecutorContext(swift::Job*) + 244
9  libswift_Concurrency.dylib     0x3c1e8 swift_job_runImpl(swift::Job*, swift::ExecutorRef) + 72
10 libdispatch.dylib              0x15164 _dispatch_root_queue_drain + 396
11 libdispatch.dylib              0x1596c _dispatch_worker_thread2 + 164
12 libsystem_pthread.dylib        0x1080 _pthread_wqthread + 228
13 libsystem_pthread.dylib        0xe5c start_wqthread + 8
```
**Additional context**

**Comment by NachoSoto** [comment]
Thanks for the detailed report! We've identifier the issue and submitted a PR: #1686.

**Comment by NachoSoto** [comment]
The fix the this will be in version 4.6.1 coming out later today :)

**Comment by quantamrhino** [comment]
You are awesome! Thank you!
